src/system/eip7251.rs

```diff
@@ -50,10 +50,7 @@ impl<Ext, Db: Database + DatabaseCommit> EvmNeedsTx<'_, Ext, Db> {
     /// consolidation requests.
     ///
     /// [EIP-7251]: https://eips.ethereum.org/EIPS/eip-7251
-    pub fn apply_eip7251(&mut self) -> Result<Bytes, EVMError<Db::Error>>
-    where
-        Db: Database + DatabaseCommit,
-    {
+    pub fn apply_eip7251(&mut self) -> Result<Bytes, EVMError<Db::Error>> {
```

Comment:
I see, so we're relaxing the bounds on the impl by moving `DatabaseCommit` strictly to the functions where it's needed. This imo makes sense to make and makes trevm easier to instantiate.